src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -78,44 +90,29 @@ public void ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled(bool
             }
-
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX) || GetAllowedProgramsToRun().Any(program => IsProgramInstalled(program)))
             {
-                string[] allowedProgramsToRun = { "xdg-open", "gnome-open", "kfmclient" };
-                foreach (var program in allowedProgramsToRun)
+                using (var px = Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = fileToOpen }))
                 {
-                    if (IsProgramInstalled(program))
+                    Assert.NotNull(px); // on Windows px is null for some reason
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
                     {
-                        programToOpen = program;
-                        break;
+                        // Assert.Equal(programToOpenWith, px.ProcessName); // on OSX, process name is dotnet for some reason. Refer to #23972
+                        Console.WriteLine($"{nameof(ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled)}(isFolder: {isFolder}), ProcessName: {px.ProcessName}");
                     }
+                    else
+                    {
+                        Assert.Equal(programToOpen, px.ProcessName);
+                    }
+                    px.Kill();
+                    px.WaitForExit();
+                    Assert.True(px.HasExited);
+                    Assert.Equal(ExitCodeKill, px.ExitCode);
                 }
-                if (programToOpen == null)
-                {
-                    Console.WriteLine($"None of the following programs were installed on this machine: {string.Join(",", allowedProgramsToRun)}.");
-                    Win32Exception e = Assert.Throws<Win32Exception>(() => Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = fileToOpen }));
-                }
-            }
-            
-            using (var px = Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = fileToOpen }))
-            {
-                Assert.NotNull(px); // on Windows px is null for some reason
-                if (!RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
-                {
-                    Assert.Equal(programToOpen, px.ProcessName);
-                }
-                else
-                {
-                    // Assert.Equal(programToOpenWith, px.ProcessName); // on OSX, process name is dotnet for some reason. Refer to #23972
-                    Console.WriteLine($"{nameof(ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled)}(isFolder: {isFolder}), ProcessName: {px.ProcessName}");
-                }
-                px.Kill();
-                px.WaitForExit();
-                Assert.True(px.HasExited);
-                Assert.Equal(ExitCodeKill, px.ExitCode);
             }
         }
-
+        
```

Comment:
extra spacing. not something we're super picky about, but it causes extra diffs in github, git diff, etc.
on my console, it shows up as red blocks when I do git diff.